### DISCUSSION TITLE
What is the file that connecting device to get socket for control device?

### DISCUSSION THREAD

**Discussion by tiennguyen12g** [yume-chan/ya-webadb#521] [General]
Hello everyone, I am new with Scrcpy, I check some folders but cannot find out the file that use to connect socket for control device. If you know can you help me? 
Thank all you.

**Comment by yume-chan** [comment]
It's the official Scrcpy server binary, downloaded from GitHub releases by this script: https://github.com/yume-chan/ya-webadb/blob/main/libraries/scrcpy/scripts/fetch-server.cjs

**Comment by yume-chan** [comment]
If you mean the client implementation, the entry point is this file: https://github.com/yume-chan/ya-webadb/blob/main/libraries/scrcpy/src/adb/client.ts

**Reply by tiennguyen12g** [reply]
correct the file that im searching. Thank bro. i will study it, if i have issue with itt, i hope you can help.
